pkg/infra/spot.go

```diff
@@ -37,6 +38,7 @@ var _ = Describe("[Feature:Machines] Running on Spot", func() {
 		platform = clusterInfra.Status.PlatformStatus.Type
 		switch platform {
 		case configv1.AWSPlatformType:
+		case configv1.GCPPlatformType:
```

Comment:
Nit, I'd keep these as a list
```suggestion
		case configv1.AWSPlatformType, configv1.GCPPlatformType:
```